File: GuessTheNumber/Core/Models/Game.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Core.Models.BaseEntity;
using Core.Models.Identity;

namespace Core.Models
{
    public class Game : BaseEntity<Guid>
    {
        public bool IsFinished { get; set; }

        [Range(long.MinValue, long.MaxValue, ErrorMessage = "The number is too long")]
        public long GuessedNumber { get; set; }

        public Guid? WinnerId { get; set; }

        public ApplicationUser Winner { get; set; }

        public Guid HostId { get; set; }

        public ApplicationUser Host { get; set; }

        public DateTimeOffset EndTime { get; set; }

        public DateTimeOffset StartTime { get; set; }

        public int PlayersCount { get; set; }

        public IList<Step> Steps { get; set; }
    }
}

File: GuessTheNumber/client-app/src/app/models/game.ts
import { Id } from "react-toastify";

export interface GameResultShort {
    id : string,
    winner: string | null,
    host: string,
    playersCount: number,
    startTime: Date,
    endTime: Date,
    guessedNumber: number
}
 
export interface GameResultFull{
    
}

export interface GameForm{
    guessedNumber: number
}
export interface GameDto{
    guessedNumber: string,
    hostId: string,
    winnerId: string,
    playersId: string[],
    isFinished: boolean,
    steps: Step[]
}

export interface Step{

}
 